k2/csrc/test_utils.h

```diff
@@ -53,33 +53,37 @@ bool ApproxEqual(FloatType a, FloatType b, double delta = 0.001) {
 }
 
 template <typename T>
-void ExpectEqual(const std::vector<T> &expected, const std::vector<T> &actual) {
+void ExpectEqual(const std::vector<T> &expected, const std::vector<T> &actual,
+                 double abs_error = 0.001) {
```

Comment:
`abs_error` is never used.

Reply:
It's template function, I used it in below float and double version

Reply:
template functions support overload.

```cpp
template<typename T>
void ExpectEqual(const std::vector<T> &expected, const std::vector<T> &actual);

template<typename T>
void ExpectEqual(const std::vector<T> &expected, const std::vector<T> &actual, T abs_error);
```
should be fine.

Reply:
mm..thinking about how you will call the two versions, you must use a if-else based on T. for simplity, let's keep the current version...but you remind me to add documentation here to say the parameter just for float and double.